File: springapp/src/main/java/com/examly/springapp/service/ScheduleTourService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.model.ScheduleTour;

public interface ScheduleTourService {

    public ScheduleTour addTour(ScheduleTour tour);

    public List<ScheduleTour> getAllTours();

    public List<ScheduleTour> getToursByStatus(String status);
}

File: angularapp/src/app/services/schedule-tour.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Global } from '../global';
import { ScheduleTour } from '../models/schedule-tour.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ScheduleTourService {

  constructor(private http: HttpClient) {}

  public addTour(tour: ScheduleTour): Observable<ScheduleTour> {
    return this.http.post(Global.baseUrl + "/api/tours", tour) as Observable<ScheduleTour>;
  }

  public getAllTours(): Observable<ScheduleTour[]> {
    return this.http.get(Global.baseUrl + "/api/tours") as Observable<ScheduleTour[]>;
  }

  public getToursByStatus(status: string): Observable<ScheduleTour[]> {
    return this.http.get(Global.baseUrl + "/api/tours/status?status=" + status) as Observable<ScheduleTour[]>;
  }
}